src/Illuminate/Routing/Router.php

```diff
@@ -1436,6 +1436,24 @@ public static function uniqueMiddleware(array $middleware)
         return $result;
     }
 
+    /**
+     * Parse an array of middleware.
+     *
+     * @param  array  $middleware
+     * @return array
+     */
+    public function parseMiddleware($middleware)
+    {
+        return Collection::make($middleware)
+            ->map(fn ($value, $key) => is_int($key)
+                ? (string) $value
+                : $key.':'.Collection::wrap($value)
+                    ->map(fn ($v) => $v === false ? '0' : $v)
```

Comment:
This is a bit of an opinion I've put in to make the casting of `true` and `false` consistent.

Usually:

```php
(string) true;

// "1"

(string) false;

// ""
```

I wanted `false` to be cast to a zero for consistency.

Reply:
I removed this. After some thought I don't think it shouldn't act differently.